src/openforms/forms/api/serializers/logic/fields.py

```diff
@@ -0,0 +1,30 @@
+import uuid
+
+from rest_framework.fields import UUIDField
+
+
+class ActionFormStepUUIDField(UUIDField):
+    """
+    UUID Field that checks the validity of the UUID but accepts empty strings (since not all rules actions need to have
+    a UUID specified for the form_step field.
+
+    It also doesn't convert the UUID to a UUID object, since this is not JSON serializable and the action is a JSON
+    field.
+    """
+
+    def to_internal_value(self, data):
```

Comment:
looks like a lot of code is taken from drf - can't we lift more on their implementation?

Something like:

```py
value = super().to_internal_value(data)
return str(value)
```

alternatively, you can also specify a different JSON encoder on the model field, and then you can use django's JSONEncoder which understands the UUID type

(I'd rather not do that now because of backporting migrations woes, but a separate PR would maybe be in order for that?)

Reply:
There is still the problem of the empty UUIDs, so I will go for the first option!